kale/pipeline/base_trainer.py

```diff
@@ -0,0 +1,168 @@
+# =============================================================================
+# Author: Xianyuan Liu, xianyuan.liu@outlook.com
+#         Haolin Wang, LWang0101@outlook.com
+# =============================================================================
+
+"""Classification of data
+The main structure is borrowed from kale.pipeline.domain_adapter.
+"""
+
+
+import pytorch_lightning as pl
+import torch
+
+from kale.predict import losses
+
+
+class BaseTrainer(pl.LightningModule):
+    """
+    Base class for classification models, based on pytorch lightning wrapper.
+    If you inherit from this class, a forward pass function must be implemented.
+    The structure is borrowed from kale.pipeline.domain_adapter.BaseAdaptTrainer.
+
+    Args:
+        optimizer (dict, None): optimizer parameters.
+        max_epochs (int): maximum number of epochs.
+        init_lr (float): initial learning rate. (default: 0.001)
+        adapt_lr (bool): adapt learning rate during training. (default: :obj:`False`)
+    """
+
+    def __init__(self, optimizer, max_epochs, init_lr=0.001, adapt_lr=False):
+        super(BaseTrainer, self).__init__()
+        self._optimizer_params = optimizer
+        self._max_epochs = max_epochs
+        self._init_lr = init_lr
+        self._adapt_lr = adapt_lr
+
+    def forward(self, x):
+        """
+        Override this function to define the forward pass. Same as :meth:`torch.nn.Module.forward()`.
```

Comment:
":meth:`torch.nn.Module.forward()`" is not very clear to me. Can you improve the readability?

Reply:
":meth:" Removed.